otx/algorithms/detection/adapters/mmdet/task.py

```diff
@@ -0,0 +1,829 @@
+
+        # Update recipe with caching modules
+        self._update_caching_modules(self._recipe_cfg.data)
+
+        logger.info("initialized.")
+
+    @staticmethod
+    def _update_anchors(origin, new):
+        logger.info("Updating anchors")
+        origin["heights"] = new["heights"]
+        origin["widths"] = new["widths"]
+
+    # pylint: disable=too-many-arguments
+    def configure(
+        self,
+        model_cfg,
+        model_ckpt,
+        data_cfg,
+        training=True,
+        subset="train",
+        ir_options=None,
+        data_classes=None,
+        model_classes=None,
+    ):
+        """Patch mmcv configs for OTX detection settings."""
+        if self._train_type == TrainType.INCREMENTAL:
+            configurer = IncrDetectionStage()
+        elif self._train_type == TrainType.SEMISUPERVISED:
+            configurer = SemiSLDetectionStage()
+        else:
+            configurer = DetectionStage()
+        cfg = configurer.configure(
+            model_cfg, model_ckpt, data_cfg, training, subset, ir_options, data_classes, model_classes
+        )
+        return cfg
+
+    # pylint: disable=too-many-branches, too-many-statements
+    def _train_model(
+        self,
+        dataset: DatasetEntity,
+    ):
+        """Train function in DetectionTrainTask."""
+        logger.info("init data cfg.")
+        self._data_cfg = ConfigDict(data=ConfigDict())
+
+        for cfg_key, subset in zip(
+            ["train", "val", "unlabeled"],
+            [Subset.TRAINING, Subset.VALIDATION, Subset.UNLABELED],
+        ):
+            subset = get_dataset(dataset, subset)
+            if subset and self._data_cfg is not None:
+                self._data_cfg.data[cfg_key] = ConfigDict(
+                    otx_dataset=subset,
+                    labels=self._labels,
+                )
+
+        # Temparory remedy for cfg.pretty_text error
+        for label in self._labels:
+            label.hotkey = "a"
+
+        self._is_training = True
+
+        if bool(self._hyperparams.tiling_parameters.enable_tiling) and bool(
+            self._hyperparams.tiling_parameters.enable_adaptive_params
+        ):
+            adaptive_tile_params(self._hyperparams.tiling_parameters, dataset)
+
+        self._init_task()
+
+        # deepcopy all configs to make sure
+        # changes under MPA and below does not take an effect to OTX for clear distinction
+        recipe_cfg = deepcopy(self._recipe_cfg)
+        data_cfg = deepcopy(self._data_cfg)
+        assert recipe_cfg is not None, "'recipe_cfg' is not initialized."
+
+        # update model config -> model label schema
+        data_classes = [label.name for label in self._labels]
+        model_classes = [label.name for label in self._model_label_schema]
+
+        recipe_cfg.work_dir = self._output_path
+        recipe_cfg.resume = self._resume
+
+        cfg = self.configure(recipe_cfg, self._model_ckpt, data_cfg, True, "train", None, data_classes, model_classes)
```

Comment:
This will probably be the final configuration used for training. Is there any code to update the Task object to have this so that it can be checked after training? I need a way to get the actual configuration used for training after training is over.

Reply:
The cfg goes to the train_detector function. Therefore I think we can dump this file

Reply:
It would be nice to have this as an attribute of the task for use in the CLI.